src/models.zig

```diff
@@ -149,6 +161,191 @@ pub const Model = struct {
+		defer tris.deinit();
+
+		var quadFaces = main.List(Quad).init(main.stackAllocator);
+		defer quadFaces.deinit();
+
+		var fixed_buffer = std.io.fixedBufferStream(data);
+		var buf_reader = std.io.bufferedReader(fixed_buffer.reader());
+		var in_stream = buf_reader.reader();
+		var buf: [128]u8 = undefined;
+		while (in_stream.readUntilDelimiterOrEof(&buf, '\n') catch |e| blk: {
+			std.log.err("Error reading line while loading model: {any}", .{e});
+			break :blk null;
+		}) |lineUntrimmed| {
+			if (lineUntrimmed.len < 3)
+				continue;
+			
+			var line = lineUntrimmed;
+			if (line[line.len - 1] == '\r') {
+				line = line[0..line.len - 1];
+			}
+			
+			if (line[0] == '#')
+				continue;
+			
+			if (std.mem.eql(u8, line[0..2], "v ")) {
+				var coordsIter = std.mem.split(u8, line[2..], " ");
+				var coords: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					coords[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				const coordsCorrect: Vec3f = .{coords[0] + 0.5, -coords[2] + 0.5, coords[1]};
+				vertices.append(coordsCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vn ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var norm: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					norm[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				const normCorrect: Vec3f = .{norm[0], -norm[2], norm[1]};
+				normals.append(normCorrect);
+			} else if (std.mem.eql(u8, line[0..3], "vt ")) {
+				var coordsIter = std.mem.split(u8, line[3..], " ");
+				var uv: Vec2f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					uv[i] = std.fmt.parseFloat(f32, coord) catch |e| blk: { std.log.err("Failed parsing {s} into float: {any}", .{coord, e}); break :blk 0; };
+				}
+				uv[0] *= 4;
+				uv[1] *= 4;
+				uvs.append(.{uv[0], uv[1]});
+			} else if (std.mem.eql(u8, line[0..2], "f ")) {
+				const n = std.mem.count(u8, line[2..], " ") + 1;
+				switch (n) {
+					3 => {
+						var coordsIter = std.mem.split(u8, line[2..], " ");
+						var faceData: [3][3]usize = undefined;
+						var i: usize = 0;
+						var failed = false;
+						while (coordsIter.next()) |vertex| : (i += 1) {
+							var d = std.mem.split(u8, vertex, "/");
+							var j: usize = 0;
+							if (std.mem.count(u8, vertex, "/") != 2 or std.mem.count(u8, vertex, "//") != 0) {
+								failed = true;
+								std.log.err("Failed loading face {s}. Each vertex must use vertex/uv/normal", .{line, n});
+								break;
+							}
+							while (d.next()) |value| : (j += 1) {
+								faceData[j][i] = std.fmt.parseUnsigned(usize, value, 10) catch |e| blk: { std.log.err("Failed parsing {s} into uint: {any}", .{value, e}); break :blk 1; };
```

Comment:
Once again, please check that these indices are in the range of the vertex/uv/normal lists.